### PR TITLE
New Qari design with play all (WIP)

### PR DESCRIPTION
**Pull request by sabeurthabti**
- [x]  Play All
- [x] Repeat
- [x] Continuous play
- [ ] Qari Landing page

screen shot 2016-09-11 at 00 45 53

@ahmedre / @mmahalwy do we currently have a way to download all audio? do we want to allow users to download all at once?

### CODE REVIEW

**Comment by mmahalwy** [comment]
@sabeurthabti no no, I don't think they should download all at once - might lock their browser.

**Comment by mmahalwy** [comment]
this is awesome!

**Comment by sabeurthabti** [comment]
screen shot 2016-09-25 at 01 21 29

**Comment by ahmedre** [comment]
"download all" used to download a torrent file (still does, but our tracker is down/broken). i'd re-set it up if we want (or you can just remove it for now).

**Comment by sabeurthabti** [comment]
@ahmedre Hmm, lets just remove it for now, and explore that option later.

**Comment by sabeurthabti** [comment]
Update:

Mobile design, suggestion:
screen shot 2016-10-06 at 00 57 42

**Comment by sabeurthabti** [comment]
Mobile design suggestion:

mobile-desk

**Comment by sabeurthabti** [comment]
- [ ]  Play All
- [ ] Repeat
- [ ] Continues play
- [ ] Qari Landing page

File: src/components/Audioplayer/index.js [hunk 1]
```diff
@@ -24,29 +26,37 @@ const styles = require('./style.scss');
...
+    shouldContinuous: state.audioplayer.shouldContinuous,
+    shouldRandom: state.audioplayer.shouldRandom,
   }),
-  { load, play, pause, playPause, update, repeat, next, previous }
+  { load, play, pause, playPause, update, repeat, next, previous, continuous, random }
 )
 export default class Audioplayer extends Component {
   static propTypes = {
     load: PropTypes.func.isRequired,
+    surahs: PropTypes.object.isRequired,
```
Comment (atalebagha) [inline src/components/Audioplayer/index.js]: I strongly believe we should use `PropTypes.shape(object)`

File: src/containers/Qari/index.js [hunk 2]
```diff
@@ -22,39 +21,61 @@ const styles = require('./style.scss');
...
     const { surahs, qari, files } = this.props;
-    const image = require(`../../../static/images/background/${(parseInt(qari.id, 10) + 1) % 33}.jpeg`);
 
     return (
       <div>
         <Grid
           fluid
-          className={`${styles.reciterBackground} ${qari.sectionId === 2 && styles.meccaBg}`}
-          style={{background: `url(${image}) center top no-repeat`}}>
+          className={`${styles.reciterBackground}`}>
```
Comment (atalebagha) [inline src/containers/Qari/index.js]: does this need a template string?

File: src/containers/Qari/index.js [hunk 3]
```diff
@@ -22,39 +21,61 @@ const styles = require('./style.scss');
...
     files: PropTypes.object.isRequired,
-    load: PropTypes.func.isRequired
+    load: PropTypes.func.isRequired,
+    play: PropTypes.func.isRequired,
+    next: PropTypes.func.isRequired,
+    continuous: PropTypes.func.isRequired,
+    Playing: PropTypes.bool.isRequired
   };
 
+  handlePlayAll() {
```
Comment (atalebagha) [inline src/containers/Qari/index.js]: Could you make this into an arrow function, there won't be any need to `bind(this)` below with an arrow function.  Also to maintain consistency with the `handleSurahSelection` function

File: src/styles/variables.scss [hunk 4]
```diff
@@ -4,14 +4,15 @@ $brand-success: #5cb85c;
...
 
 $font-size-base: 14px;
 $font-family-sans-serif: 'Montserrat', sans-serif;
+$font-family-regular: 'Montserrat-regular', sans-serif;
 
 $navbar-inverse-bg: transparent;
 $navbar-inverse-color: #fff;
 $navbar-inverse-link-color: #fff;
-
 @import "node_modules/bootstrap-sass/assets/stylesheets/bootstrap/variables";
```
Comment (mmahalwy) [inline src/styles/variables.scss]: That was my addition. Cause I want to use a bootstrap variable within our variables.

Comment (mmahalwy) [inline src/styles/variables.scss]: Probably not a good idea

**Comment by sabeurthabti** [comment]
@atalebagha I will review your feed these tonight (GMT), thanks.

File: src/components/Audioplayer/index.js (hunk 1 above)
Comment (sabeurthabti) [inline src/components/Audioplayer/index.js]: agreed

Comment (sabeurthabti) [inline src/components/Audioplayer/index.js]: Actually this is harder than I expected, surahs is an object of all 114 surahs. I added Shape on the surah which is more realistic.

File: src/components/Audioplayer/index.js [hunk 5]
```diff
@@ -172,7 +189,7 @@ export default class Audioplayer extends Component {
     const { shouldRepeat, repeat } = this.props; // eslint-disable-line no-shadow
 
     return (
-      <div className={`text-center pull-right ${styles.repeat} ${shouldRepeat && styles.active}`}>
+      <div className={`text-center pull-right ${styles.toggle} ${shouldRepeat && styles.active}`}>
```
Comment (sabeurthabti) [inline src/components/Audioplayer/index.js]: agreed, apply for you to add this. I tried and failed haha.

**Pull request merged**